### DISCUSSION TITLE
Feature request: find orphaned dependencies

### DISCUSSION THREAD

**Discussion by BackEndTea** [composer-unused/composer-unused#75] [Ideas]
## Is your feature request related to a problem? Please describe
As described here: https://twitter.com/icanhazstring/status/1227190874855165953

I require package `A`. `A` has a dependency on package `B`.
Later i remove package `A`, but `B` remains in my `composer.lock`. This can be due to manually editing the composer.lock, merge conflicts, or other reasons.

B is now an orphaned dependency, and using composer why for it even says its not installed. 

## Describe the solution you'd like
I would like composer unused to find these dependencies, maybe with another command.

## Describe alternatives you've considered
I wrote my own 'quick and dirty' implementation right [here](https://gist.github.com/BackEndTea/525f96e339abf443c47dfd884e222c15), which make a graph of the dependencies, and does a (rather unoptimized) DFS to check what dependencies aren't part of the main graph.  

## Teachability, Documentation, Adoption, Migration Strategy

This should probably be a new command.

**Comment by icanhazstring** [comment]
Hi @BackEndTea.

I was just trying to replicate the problem.
What I have done was:

```
$ composer require phpunit/phpunit
$ composer remove phpunit/phpunit --no-update-with-dependencies
```

Which leaves a lot of dependencies from phpunit left in my system.
A simple `composer udpdate` would do the trick, but I guess this is not what you wanted, since it updates also everything else.

With your provided script, you simple scan of any of the installed packages are linked to one of you root requirements right? If not, the will be removed.

Am I correct?

**Comment by BackEndTea** [comment]
> With your provided script, you simple scan of any of the installed packages are linked to one of you root requirements right? If not, the will be removed.

Yes that is how it works.
